doc/tools/create_changelog.py

```diff
@@ -0,0 +1,161 @@
+#!/usr/bin/python
+from __future__ import absolute_import
+from __future__ import unicode_literals
+from __future__ import print_function
+import sys
+import io
+from subprocess import check_output, CalledProcessError
+import json
+try:
+    from urllib import urlencode
+except ImportError:
+    from urllib.parse import urlencode
+try:
+    from urllib2 import Request, urlopen
+except ImportError:
+    from urllib.request import Request, urlopen
+
+# this date needs to be updated with the date of the previous release
+LastReleaseDate = '2014-9-30'
+
+# this probably won't change
+RepoName = "NREL/EnergyPlus"
+EPlusRepoPath = 'https://github.com/' + RepoName
+
+
+def usage():
+    print("""Script should be called with 4 positional arguments:
+ - the path to a repository 
+ - the path to a markdown output file
+ - the path to a html output file
+ - the path to a local git executable
+ - a github token for performing authentication API requests
+ - and optionally a "Y" for enabling debug mode""")
+
+# command line arguments: repo base path, output markdown and html file paths, a git exe path, and a github token
+if len(sys.argv) == 6:
+    repo = sys.argv[1]
+    md_file = sys.argv[2]
+    html_file = sys.argv[3]
+    git_exe = sys.argv[4]
+    github_token = sys.argv[5]
+    debug = False
+elif len(sys.argv) == 7:
+    repo = sys.argv[1]
+    md_file = sys.argv[2]
+    html_file = sys.argv[3]
+    git_exe = sys.argv[4]
+    github_token = sys.argv[5]
+    if sys.argv[6] == "Y":
+        debug = True
+    else:
+        print("Bad debug flag, should be \"Y\" or not passed at all")
+        usage()
+        sys.exit(1)
+else:
+    usage()
+    sys.exit(1)
+
+# get the pull request numbers
+try:
+    log_full = check_output([git_exe, 'log', '--oneline', '--after=' + LastReleaseDate]).decode('utf-8')
+except CalledProcessError as ex:
+    log_full = ''
+    pass  # add error handling
+log_full_split = log_full.split('\n')
+log_merge_prs = [x for x in log_full_split if 'Merge pull request' in x]
```

Comment:
This seems to be assuming commit format.  What if they were not merged from github?  How do you intervene with this system to capture things in changlog that might not be tracked through PR/commits?